nuxeo-java-client/src/main/java/org/nuxeo/client/api/objects/Document.java

```diff
@@ -278,6 +289,9 @@ public String getName() {
     }
 
     public void set(String key, Object value) {
+        if (value instanceof Calendar || value instanceof Date) {
```

Comment:
Yes, the `Document#set` is a general method for all the properties.

> On the other hand you could convert from Calendar and Date to the proper String format expected

This was my initially propose. However, if I did a conversion inside the `set` method, it would lead to 2 undesired results:

1. The auto-conversion only happens to `set`. Using other setter method will not hook the conversion, such as the multi-params constructor and the method `setProperties(Map<String, Object>)`. So users might be confused about the different behaviors among different setters.

2. When people want the retrieve the key after document update, the result will not be consistent. 

    ```java
    Calendar cal = new Calendar();
    doc.set('dc:issued', cal);  // auto-conversion happens inside the setter
    ...
    // after update
    cal = doc.get('dc:issued'); // won't work
    ```

    That's why I dion't want to handle the conversion.